File: TravelAgencyFrontend/src/app/types/Reservation/ReservationResponse.ts
import {PaymentStatus} from "./PaymentStatus";
import {PaymentType} from "./PaymentType";
import {TravelResponse} from "../Travel/TravelResponse";
import {Pack} from "../Pack/Pack";
import {Hotel} from "../Hotel/Hotel";
import {ClientResponse} from "../Client/ClientResponse";
import {CompanionResponse} from "../Companion/CompanionResponse";

export interface ReservationResponse {
  id?: string;
  paymentStatus: PaymentStatus;
  paymentType: PaymentType;
  travel: TravelResponse;
  pack: Pack;
  discount?: number;
  companions: CompanionResponse[];
  hotel: Hotel;
  hosting?: string;
  client: ClientResponse;
  travelPrice: number;
}

File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/response/ReservationResponse.java
package com.travelagency.travelagencyapplication.dto.response;

import com.travelagency.travelagencyapplication.collection.*;
import com.travelagency.travelagencyapplication.dto.request.PackRequest;
import com.travelagency.travelagencyapplication.enums.PaymentStatus;
import com.travelagency.travelagencyapplication.enums.PaymentType;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.DBRef;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReservationResponse {

    private String id;

    private PaymentStatus paymentStatus;

    private PaymentType paymentType;

    private TravelResponse travel;

    private PackResponse pack;

    private Double discount;

    private Set<CompanionResponse> companions;

    private HotelResponse hotel;

    private String hosting;

    private ClientResponse client;

    private Double travelPrice;
}
